opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1029,12 +1027,15 @@ namespace Opm {
                 const unsigned cellIdx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& intQuants = elemCtx.intensiveQuantities(/*spaceIdx=*/0, /*timeIdx=*/0);
                 const auto& fs = intQuants.fluidState();
+                const double pv =
+                    ebosSimulator_.model().dofTotalVolume(cellIdx)
+                    * intQuants.porosity().value();
```

Comment:
comment added. IMO the second point is self-evident because problem.porosity() returns a scalar instead of an evaluation. (i.e., it cannot be solution dependent.)